addons/l10n_mx/views/account_move_view.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_move_form_l10n_mx_inherit" model="ir.ui.view">
+        <field name="name">account.move.form.l10n_mx</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form"/>
+        <field name="arch" type="xml">
+            <field name="tax_totals" position="attributes">
+                <attribute name="readonly">state != 'draft' or (move_type not in ('in_invoice', 'in_refund', 'in_receipt') and not quick_edit_mode and not tax_totals['local_tax_edit_mode'])</attribute>
```

Comment:
You cannot change a generic behavior when installing a localization. Here, you are changing it for everyone.
Also, it doesn't work if you have a local tax mixed with a not-local tax in the same tax group.
I would be cleaner to add generically a way in the tax totals to decide if the tax group is editable or not.
In your case, the tax group will be editable only in all involved taxes are mexican local taxes